timelock-impl/src/main/java/com/palantir/atlasdb/timelock/paxos/AutobatchingPaxosPingableLeaderFactory.java

```diff
@@ -0,0 +1,91 @@
+/*
+ * (c) Copyright 2019 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.atlasdb.timelock.paxos;
+
+import java.io.Closeable;
+import java.util.UUID;
+import java.util.concurrent.ExecutionException;
+
+import com.palantir.atlasdb.autobatch.Autobatchers;
+import com.palantir.atlasdb.autobatch.Autobatchers.SupplierKey;
+import com.palantir.atlasdb.autobatch.DisruptorAutobatcher;
+import com.palantir.leader.PingableLeader;
+
+public class AutobatchingPaxosPingableLeaderFactory implements Closeable {
```

Comment:
nit: I guess it's named this way because the package has `paxos` in it, but it doesn't look like the class is specific to Paxos leaders

Reply:
fair, this was just muscle memory tbh aha